lib/foreman_docker/engine.rb

```diff
@@ -58,32 +58,41 @@ class Engine < ::Rails::Engine
 
         security_block :containers do
           permission :view_containers,
-                     :containers          => [:index, :show],
-                     :'api/v2/containers' => [:index, :show, :logs]
-          permission :commit_containers, :containers => [:commit]
+                     { :containers          => [:index, :show],
+                     :'api/v2/containers' => [:index, :show, :logs] },
+                     :resource_type => 'Container'
+          permission :commit_containers, { :containers => [:commit] },
+                     :resource_type => 'Container'
           permission :create_containers,
-                     :'containers/steps'  => [:show, :update],
+                     { :'containers/steps'  => [:show, :update],
                      :containers          => [:new],
-                     :'api/v2/containers' => [:create, :power]
+                     :'api/v2/containers' => [:create, :power] },
+                     :resource_type => 'Container'
           permission :destroy_containers,
-                     :containers          => [:destroy],
-                     :'api/v2/containers' => [:destroy]
+                     { :containers          => [:destroy],
+                     :'api/v2/containers' => [:destroy] },
+                     :resource_type => 'Container'
           permission :power_compute_resources_vms,
-                     :containers          => [:power],
-                     :'api/v2/containers' => [:create, :power]
+                     { :containers          => [:power],
+                     :'api/v2/containers' => [:create, :power] },
+                     :resource_type => 'ComputeResource'
         end
 
         security_block :registries do
-          permission :view_registries,    :registries         => [:index, :show]
-          permission :create_registries,  :registries         => [:new, :create, :update, :edit]
-          permission :destroy_registries, :registries         => [:destroy]
+          permission :view_registries, { :registries => [:index, :show] },
+                     :resource_type => 'Registry'
+          permission :create_registries, { :registries  => [:new, :create, :update, :edit] },
```

Comment:
Line is too long. [93/80]<br>Unnecessary spacing detected.<br>Use the new Ruby 1.9 hash syntax.